tests/test/slg.rs

```diff
@@ -1,147 +0,0 @@
-use crate::test_util::*;
-use chalk_integration::db::ChalkDatabase;
-use chalk_solve::ext::*;
-use chalk_solve::SolverChoice;
-
-macro_rules! test {
-    (program $program:tt $(goal $goal:tt first $n:tt with max $depth:tt { $expected:expr })*) => {
-        solve_goal(stringify!($program), vec![$(($depth, $n, stringify!($goal), $expected)),*])
-    };
-}
-
-fn solve_goal(program_text: &str, goals: Vec<(usize, usize, &str, &str)>) {
-    println!("program {}", program_text);
-    assert!(program_text.starts_with("{"));
-    assert!(program_text.ends_with("}"));
-    let db = ChalkDatabase::with(
-        &program_text[1..program_text.len() - 1],
-        SolverChoice::default(),
-    );
-    db.with_program(|_| {
-        for (max_size, num_answers, goal_text, expected) in goals {
-            println!("----------------------------------------------------------------------");
-            println!("goal {}", goal_text);
-            assert!(goal_text.starts_with("{"));
-            assert!(goal_text.ends_with("}"));
-            let goal = db
-                .parse_and_lower_goal(&goal_text[1..goal_text.len() - 1])
-                .unwrap();
-            let peeled_goal = goal.into_peeled_goal();
-            let mut slg_solver = SolverChoice::SLG {
-                max_size,
-                max_answers: None,
-            }
-            .into_solver()
-            .into_test();
-            let result = format!(
-                "{:#?}",
-                slg_solver.force_answers(&db, &peeled_goal, num_answers)
-            );
-            // Strip trailing commas to handle both nightly and stable debug formatting
-            let result = result.replace(",\n", "\n");
-            let expected = expected.replace(",\n", "\n");
-            assert_test_result_eq(&expected, &result);
-        }
-    });
-}
-
-#[test]
-fn subgoal_cycle_uninhabited() {
```

Comment:
Why remove this test?

Reply:
Ah yeah, you wrote this:

> One point to ask about: the subgoal_cycle_uninhabited test is now giving me a well-formed error, either though the program is basically identical to subgoal_cycle_inhabited minus an impl.

Did you in fact move the test?

Reply:
I believe that the "SLG Solver" tests skipped the well-formedness

Reply:
It's in `misc` now :) However, it's ignored because it's giving me a well-formed error and wanted to see if you could figure out why.